node/src/components/deploy_buffer.rs

```diff
@@ -116,10 +129,38 @@ impl DeployBuffer {
     pub(crate) fn new<REv>(
         registry: &Registry,
         effect_builder: EffectBuilder<REv>,
+        linear_chain: &[Block],
     ) -> Result<(Self, Effects<Event>), prometheus::Error>
     where
         REv: ReactorEventT,
     {
+        let mut effects = effect_builder
+            .set_timeout(DEPLOY_BUFFER_PRUNE_INTERVAL)
+            .event(|_| Event::BufferPrune);
+
+        for block in linear_chain.iter() {
+            let deploy_hashes = block.deploy_hashes();
+            let mut hashes = SmallVec::with_capacity(deploy_hashes.len());
+            let block_hash =
+                ProtoBlockHash::from_parts(&deploy_hashes, block.header().random_bit());
+            hashes.extend(deploy_hashes.iter().cloned());
+            effects.extend(
+                effect_builder
+                    .get_deploys_from_storage(hashes)
+                    .event(move |result| {
+                        let deploys = result
+                            .iter()
+                            .flatten()
+                            .map(|deploy| (*deploy.id(), deploy.header().clone()))
+                            .collect::<HashMap<_, _>>();
+                        Event::GetDeploysForFinalizedBlockResult {
+                            block_hash,
+                            deploys,
+                        }
```

Comment:
There's a danger that we can't ignore that those events will be handled too late, after consensus asks deploy buffer for deploys. Why not simply create a `finalized_deploys` from the `linear_chain` blocks? There's enough data to reconstruct it, right?